File: src/main/java/no/stunor/origo/eventorapi/services/converter/ContactConverter.java
package no.stunor.origo.eventorapi.services.converter;

import java.util.List;

import org.iof.eventor.Tele;

public class ContactConverter {
 
      public static String convertEmail(List<Tele> teleList) {
        if(teleList == null){
            return null;
        }
        for(Tele tele : teleList){
            if(tele.getTeleType().getValue().equals("official")){
                if(tele.getMailAddress() != null){
                    return tele.getMailAddress();
                }
            }
        }
        return null;
    }

    public static String convertPhone(List<Tele> teleList) {
        if(teleList == null){
            return null;
        }
        for(Tele tele : teleList){
            if(tele.getTeleType().getValue().equals("official")){
                if(tele.getMobilePhoneNumber() != null){
                    return tele.getMobilePhoneNumber();
                }
            }
        }
        return null;
    }
}

File: src/main/kotlin/no/stunor/origo/eventorapi/services/converter/ContactConverter.kt
package no.stunor.origo.eventorapi.services.converter

import org.iof.eventor.Tele
import org.springframework.stereotype.Component

@Component

class ContactConverter {
    fun convertEmail(teleList: List<Tele>): String? {
        for (tele in teleList) {
            if (tele.teleType.value == "official" && tele.mailAddress != null) {
                return tele.mailAddress
            }
        }
        return null
    }

    fun convertPhone(teleList: List<Tele>): String? {
        for (tele in teleList) {
            if (tele.teleType.value == "official" && tele.mobilePhoneNumber != null) {
                return tele.mobilePhoneNumber
            }
        }
        return null
    }
}
